### PR TITLE
More `flake8-bugbear`: fix B014, B017

### PR DESCRIPTION
**Pull request by mattwthompson**
Another  follow-on to https://github.com/openforcefield/openff-toolkit/pull/1539; in this case `pytest.raises(Exception)` was masking some (in my opinion) subpar error handling, so I implemented some more discrete cases.

A couple of the other smaller changes would have been caught by https://github.com/asottile/pyupgrade, a tool I've used for a long time and think does a good job of keeping old (i.e. Python 2 and <3.8) constructs out.  It's easy to add - don't need to do that now.

### CODE REVIEW

File: openff/toolkit/utils/openeye_wrapper.py [hunk 1]
```diff
@@ -203,7 +203,7 @@ def is_available(cls) -> bool:
                     cls._is_installed = True
                     try:
                         importlib.import_module("openeye." + tool)
-                    except (ImportError, ModuleNotFoundError):
+                    except ImportError:
```
Comment (j-wags) [inline openff/toolkit/utils/openeye_wrapper.py:206]: I can't manage to get `importlib.import_module` to raise an `ImportError` - All of the following give me `ModuleNotFoundError`:

* `module = importlib.import_module("openeye.oechem")` raises no error, even when missing an OE license
* `module = importlib.import_module("openeye.zzz")` gives a `ModuleNotFoundError`
* `module = importlib.import_module("zzzz")` gives a `ModuleNotFoundError`

But I'm also not confident that this won't raise an `ImportError` - There's a lot of voodoo around python imports and OE Toolkit performance that I don't have a lot of confidence about. 

I'd propose that we revert these ones to `except (ImportError, ModuleNotFoundError):` and live with the bugbear we know.

Comment (mattwthompson) [reply openff/toolkit/utils/openeye_wrapper.py:206]: `ModuleNotFoundError` is a subclass of `ImportError`

```python3
>>> issubclass(ModuleNotFoundError, ImportError)
True
```

Installing OpenEye Toolkits doesn't require an active license, nor does importing `openeye.oechem`; if you uninstall it that will raise a `ModuleNotFoundError`

Comment (j-wags) [reply openff/toolkit/utils/openeye_wrapper.py:206]: Ahhh, thanks for the explanation!

**Review by j-wags — APPROVED** [review]
LGTM! Thanks for the cleanups!

**Pull request merged**